lib/index.js

```diff
@@ -13,14 +13,22 @@ function compareColors(nodeColor, targetColor) {
     .every((c, i) => parseFloat(c.value) === targetColor[i]);
 }
 
-function applyColorFunction(node, initialColor, finalColor) {
+function applyColorFunction(node, initialColor, finalColor, alphaChannel) {
   const {name} = node;
   const nodeColor = node.children
     .filter(item => item.type === 'Number' || item.type === 'Percentage')
     .toArray();
 
-  if (compareColors(nodeColor, initialColor[name])) {
+  const cloneNodeColor = [...nodeColor];
+  const sliceNodeColor = cloneNodeColor.slice(0, 3);
+  const sliceInitialColor = initialColor[name].slice(0, 3);
+
+  if (!alphaChannel && compareColors(nodeColor, initialColor[name])) {
     nodeColor.forEach((c, i) => c.value = finalColor[name][i]);
+  } else if(alphaChannel && compareColors(sliceNodeColor, sliceInitialColor)) {
+    nodeColor.forEach((c, i, a) => {
```

Comment:
Also, I would implement a function to replace nodeColor components with `preserveAlphaChannel` option. So you will have consistent check and replace functions which get the same parameters.

This makes your main function a much cleaner and smaller.

Reply:
Okey